inference-engine/ie_bridges/java/sample/face_detection_sample_async/Main.java

```diff
@@ -0,0 +1,246 @@
+import org.opencv.core.*;
+import org.opencv.imgcodecs.*;
+import org.opencv.videoio.*;
+import org.opencv.videoio.VideoWriter;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.LinkedList;
+import java.util.Vector;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.Map;
+import java.util.Queue;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.intel.openvino.*;
+
+public class Main {
+ 
+    public static Blob imageToBlob(Mat image) {
+        if (buff == null) 
+            buff = new byte[(int) (image.total() * image.channels())];
+
+        image.get(0, 0, buff);
+
+        int[] dimsArr = {1, image.channels(), image.height(), image.width()};
+        TensorDesc tDesc = new TensorDesc(Precision.U8, dimsArr, Layout.NHWC);
+
+        return new Blob(tDesc, buff);
+    }
+
+    static void processInferRequets(String outputName, Queue<Integer> startedRequestsIds, Vector<Boolean> asyncInferIsFree, Vector<InferRequest> inferRequests) {
+            int requestId = startedRequestsIds.peek();
+            InferRequest inferRequest = inferRequests.get(requestId);
+
+            if (inferRequest.Wait(WaitMode.STATUS_ONLY) != StatusCode.OK)
+                return;
+
+            detectionOutput.add(inferRequest.GetBlob(outputName).asFloat());
+            resultCounter++;
+
+            asyncInferIsFree.setElementAt(true, requestId);
+            startedRequestsIds.remove();
+    }
+
+    public static void main(String[] args) {
+        try {
+            System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+            System.loadLibrary(IECore.NATIVE_LIBRARY_NAME);
+        } catch (UnsatisfiedLinkError e) {
+            System.err.println("Failed to load native library\n" + e);
+            System.exit(1);
+        }
+
+        Map<String, String> input = new HashMap<>();
+
+        try{
+            for(int i = 0; i < args.length; i++)
+                input.put(args[i], args[++i]);
+        } catch(ArrayIndexOutOfBoundsException e) {
+            System.out.println("Error: Incorrect number of arguments");
+            return;
+        }
```

Comment:
Please create a separate class like `ArgumentParser` which will receive `args` and have method `get`